src/components/Requirements/Requirements.vue

```diff
@@ -169,14 +168,23 @@ export default Vue.extend({
     },
     onShowAllCourses(showAllCourses: {
       requirementName: string;
-      subReqCoursesArray: SubReqCourseSlot[];
+      subReqCoursesArray: FirestoreSemesterCourse[];
     }) {
       this.shouldShowAllCourses = true;
-      const allPotentialCourses = showAllCourses.subReqCoursesArray
-        .flatMap(slot => (slot.isCompleted ? [] : slot.courses))
-        .slice(0, 24);
-      const lastCourse = allPotentialCourses[allPotentialCourses.length - 1];
-      this.lastLoadedShowAllCourseId = lastCourse.crseId;
+
+      // limited to 24 courses
+      const allPotentialCourses = showAllCourses.subReqCoursesArray.slice(
+        this.lastLoadedShowAllCourseId,
+        this.lastLoadedShowAllCourseId + 24
+      );
+
+      // next time see all is open, show next 24 courses (unless already reached the end)
+      if (this.lastLoadedShowAllCourseId + 24 > showAllCourses.subReqCoursesArray.length) {
+        this.lastLoadedShowAllCourseId = 0;
+      } else {
+        this.lastLoadedShowAllCourseId += 24;
+      }
+
```

Comment:
I'm not really a fan of showing 24 different courses every time the user clicks See All. I think it could be a little confusing. Could we possibly re-introduce pagination? I think a long term goal for courses with lots of See All courses (e.g. CALS) would be to have an add course button that opens up the add modal where the only options are the See All courses. So for example, if I click an add course button for CALS credits, when I start typing a course to add in the add modal, the dropdown suggestions are the See All courses. @einc would you agree?

Reply:
+1 on showing 24 different courses. is pagination possible/easy to implement? this is a super rough design and ill look into it after todays meeting but would something like this be do-able by launch?
<img width="513" alt="Screen Shot 2021-02-28 at 5 29 15 PM" src="https://user-images.githubusercontent.com/14082828/109435878-b0c5e700-79ea-11eb-8cb5-866bbd37d00d.png">

Reply:
idk if it would be possible to do it like before but could we do something like once you scroll to a certain point, more courses get loaded. And then this batching stops once you get to the end?

Reply:
@tcho6319 I think this approach still needs to render increasing number of cards. It's however possible to do virtual list rendering: calculating how many cards are visible and only render those visible and a few cards around them. However, this is much more involved, and the pagination solution is much simpler.